engines/zypper_auth/lib/zypper_auth/engine.rb

```diff
@@ -140,8 +153,23 @@ def verify_instance
         alias_method :original_path_allowed?, :path_allowed?
 
         # additional validation for strict_authentication auth subrequest
-        def path_allowed?(path)
-          return false unless original_path_allowed?(path)
+        def path_allowed?(headers)
+          return false unless original_path_allowed?(headers)
```

Comment:
What's going on here, `original_path_allowed?` is an alias for `path_allowed?`, and then we call `original_path_allowed?` from inside `path_allowed?`?

Reply:
correct

Reply:
Well, 2 questions here: 
* Why is that alias needed? 
* Why is that method recursively calling into itself?